OpenRA.Mods.Common/ServerTraits/MasterServerPinger.cs

```diff
@@ -82,12 +82,25 @@ public void Tick(S server)
 					server.SendMessage(masterServerMessages.Dequeue());
 		}
 
-		public void ServerStarted(S server)
+		void INotifyServerStart.ServerStarted(S server)
 		{
 			if (server.Type != ServerType.Local && LanGameBeacon != null)
 				LanGameBeacon.Start();
 		}
 
+		void INotifyServerShutdown.ServerShutdown(S server)
+		{
+			if (server.Settings.AdvertiseOnline)
+			{
+				// Announce that the game has ended to remove it from the list.
+				var gameServer = new GameServer(server);
+				UpdateMasterServer(server, gameServer.ToPOSTData(false));
+			}
+
+			if (LanGameBeacon != null)
```

Comment:
The null check is no longer needed - this is what the `?.` does.

Reply:
Removed the redundant check.